### DISCUSSION TITLE
Possible to have a date picker close to native ones

### DISCUSSION THREAD

**Discussion by frec-hansen** [mantinedev/mantine#6304] [Feature requests]
Current date picker is a string input field. Is it possible to have a date picker that allows user to enter MM/DD/YYYY separately like this one from the MUI: https://mui.com/x/react-date-pickers/date-picker/

Currently I can achieve something similar with the native date picker with `<input type="date" />` but the design doesn't match Mantine and the Calendar dropdown looks really ugly.

**Comment by frec-hansen** [comment]
Just saw the time picker, it would be similar but just for date

**Comment by rtivital** [comment]
https://mantine.dev/dates/date-input/

**Reply by frec-hansen** [reply]
I saw the component and played around with it. Is there any way to make it behave like this:
<img width="251" alt="Screenshot 2024-05-29 at 2 53 25 PM" src="https://github.com/mantinedev/mantine/assets/97467751/46ce0e5a-89b1-4bff-ad48-f55256f9d855">
Where user can type MM/DD/YYYY separately

**Reply by rtivital** [reply]
If you need mask feature, use – https://www.npmjs.com/package/react-imask
You can build a custom component with InputBase, Popover and Calendar for that – https://mantine.dev/core/input/#inputbase-component

**Reply by frec-hansen** [reply]
It is not really a mask feature because if you use the native `input` with type `date`, by default the MM/DD/YYYY is separated. I guess I will see if I can make some custom component. Thanks for the help.
